### PR TITLE
Social sharing

### PR DESCRIPTION
**Pull request by andreaWaldren**
**JIRA Ticket**: (link) (:star:)
https://webapps.es.vt.edu/jira/browse/LIBTD-2167

# What does this Pull Request do? (:star:)
Adds social media sharing buttons to Archive and Collection pages.

# What's the changes? (:star:)
-Adds package for react-share
-SocialButtons - Component is set up to render any of the available social media buttons allowed through the package. 
-ArchivePage and CollectionShowPage implement the buttons on these pages.
-Several CSS files changed to standardize some differences in fonts and spacing between the archive and collection pages.

# How should this be tested?
On any archive page, click the buttons to to make sure they work and share the correct info. On any collection page, click the buttons and make sure they share the current part of the collection being viewed.

# Additional Notes:
Branch is LIBTD-2167

# Interested parties
@yinlinchen 

(:star:) Required fields

### CODE REVIEW

File: src/pages/archives/ArchivePage.js [hunk 1]
```diff
@@ -289,7 +290,20 @@ class ArchivePage extends Component {
...
-              <Citation item={this.state.item} />
+              <SocialButtons
+                buttons={[
+                  "Email",
+                  "Facebook",
+                  "Twitter",
+                  "Pinterest",
+                  "Tumblr",
+                  "Reddit"
+                ]}
```
Comment (yinlinchen) [inline src/pages/archives/ArchivePage.js:301]: I can not tell from the ticket description https://webapps.es.vt.edu/jira/browse/LIBTD-2167. did anyone already decide we will use these six services?

File: src/components/SocialButtons.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
+import React, { Component } from "react";
+import {
+  EmailShareButton,
```
Comment (yinlinchen) [inline src/components/SocialButtons.js:3]: See comment below, if we decide use just that six services, do you need to include all these? It seems we will never use most of services...

**Review by yinlinchen — CHANGES_REQUESTED** [review]

File: src/components/SocialButtons.js (hunk 2 above)
Comment (andreaWaldren) [reply src/components/SocialButtons.js:3]: I think we should include all of them and in a separate ticket set up a configuration option in the admin area to let the stakeholders choose which social media to allow. I just picked 6 for now to demonstrate.

Comment (yinlinchen) [reply src/components/SocialButtons.js:3]: I don't think we should include all of them, as I said earlier, some services we will never use, so why include the codes we will never use?

I do agree we can have another ticket for configuration option.

Comment (andreaWaldren) [reply src/components/SocialButtons.js:3]: My apologies, I thought your previous message was asking for my opinion. I don't feel strongly either way about this, it was just my opinion. I just pushed changes that removes all the lesser known networks, so the options are reduced by about half.

File: src/pages/collections/CollectionsShowPage.js [hunk 3]
```diff
@@ -269,6 +275,22 @@ class CollectionsShowPage extends Component {
...
+                        "Facebook",
+                        "Twitter",
+                        "Pinterest",
+                        "Tumblr",
+                        "Reddit"
+                      ]}
+                      url={window.location.href}
+                      image={this.state.thumbnail_path}
+                      title={this.getTitle()}
+                    />
```
Comment (yinlinchen) [inline src/pages/collections/CollectionsShowPage.js:291]: Please create a components for this so both `collectionshowpage` and `archivepage` can use it.

**Review by yinlinchen — CHANGES_REQUESTED** [review]
see the lastest comment above

**Comment by andreaWaldren** [comment]
I'm sorry but I don't understand. The SocialButtons.js file is a component and it has been imported on both the archive and collection pages. Does that not show up in my changes?

File: src/pages/collections/CollectionsShowPage.js (hunk 3 above)
Comment (andreaWaldren) [reply src/pages/collections/CollectionsShowPage.js:291]: I'm sorry but I don't understand. The SocialButtons.js file is a component and it has been imported on both the archive and collection pages. Does that not show up in my changes?

Comment (yinlinchen) [reply src/pages/collections/CollectionsShowPage.js:291]: https://github.com/VTUL/dlp-access/pull/246/files#diff-c550e3639b774a9f35e9d86b555322296bd4b33ecb47778126d0a340780cb1feR293 and https://github.com/VTUL/dlp-access/pull/246/files#diff-df7761cb3b73e8f53e4c2c5fa7abd224f93027ccda65f821f19da238e5a5e901R278 can be merged and to be used by both pages.

Comment (andreaWaldren) [reply src/pages/collections/CollectionsShowPage.js:291]: I still am not 100% sure, but are you suggesting to merge the citation component and the social media buttons?

Comment (yinlinchen) [reply src/pages/collections/CollectionsShowPage.js:291]: 1. create a component like
```
class CustomSocialButton extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            buttons: []
        };
    }

    async getButtonList() {
        let buttonlist = ["Email", "Facebook", "Twitter", "Pinterest", "Tumblr", "Reddit"];
        this.setState({ buttons: buttonlist });
    }

    componentDidMount() {
        this.getButtonList();
    }

    render() {
        return  ( 
            <SocialButtons
                buttons={this.state.buttons}
                url={window.location.href}
                {...this.props}
            />
        );
    }

}
```
2. Then both collectionshowpage and archivepage can use it, like 
archivepage
```
<CustomSocialButton
  image={this.state.item.thumbnail_path}
  title={this.state.item.title}
/>
```
and collectionshowpage
```
<CustomSocialButton
  image={this.state.thumbnail_path}
  title={this.getTitle()}
/>
```

also create a JIRA ticket to implement getButtonList() that fetch configuration options

**Comment by yinlinchen** [comment]
@andreaWaldren  See https://github.com/VTUL/dlp-access/pull/246/files#r581096882

**Pull request closed without merging**